R/zchunk_LA100.IEA_downscale_ctry.R

```diff
@@ -0,0 +1,308 @@
+        prod <- L100.USSR_Yug$PRODUCT
+        prod_obc <- prod == "Other bituminous coal"
+        prod_hcind <- prod == "Hard coal (if no detail)"
+        prod_cc <- prod == "Coking coal"
+        L100.USSR_Yug[prod_obc, NO_DETAIL_COAL_YEARS] <-
+          L100.USSR_Yug[prod_hcind, NO_DETAIL_COAL_YEARS ] *
+          round(L100.USSR_Yug$`1978`[prod_obc] /
+                  (L100.USSR_Yug$`1978`[prod_cc] + L100.USSR_Yug$`1978`[prod_obc] + 1e-3), digits = 2)
+        L100.USSR_Yug[prod_cc, NO_DETAIL_COAL_YEARS ] <-
+          L100.USSR_Yug[prod_hcind, NO_DETAIL_COAL_YEARS ] - L100.USSR_Yug[prod_obc, NO_DETAIL_COAL_YEARS ]
+        L100.USSR_Yug[prod_hcind, NO_DETAIL_COAL_YEARS ] <- 0
+
+        # Brown coal is simpler, as lignite is the only relevant fuel (104-119)
+        # (sub-bituminous coal is not considered in these regions)
+        prod_bcind <- prod == "Brown coal (if no detail)"
+        L100.USSR_Yug[prod == "Lignite", NO_DETAIL_COAL_YEARS ] <- L100.USSR_Yug[prod_bcind, NO_DETAIL_COAL_YEARS ]
+        L100.USSR_Yug[prod_bcind, NO_DETAIL_COAL_YEARS ] <- 0
+      }
+
+      # Isolate 1990 data for FSU and Yugoslavia; this will be used to back-project individual country shares
+      fsu_yug_composite_entries <- IEA_ctry_composite$IEA_ctry %in% c("Former Soviet Union (if no detail)", "Former Yugoslavia (if no detail)")
+      L100.IEAsingle %>%
+        filter(iso %in% IEA_ctry_composite$iso[fsu_yug_composite_entries]) %>%
+        left_join_keep_first_only(select(IEA_ctry_composite, iso, IEA_ctry), by = "iso") %>%
+        rename(IEAcomposite = IEA_ctry) ->
+        L100.USSR_Yug_ctry
+
+      L100.USSR_Yug_ctry %>%
+        filter(paste(PRODUCT, FLOW) %in% paste(IEA_product_downscaling$PRODUCT, IEA_product_downscaling$FLOW)) ->
+        L100.USSR_Yug_ctry_FLOW_PRODUCT
+
+      L100.USSR_Yug_ctry_FLOW_PRODUCT %>%
+        group_by(IEAcomposite, FLOW, PRODUCT) %>%
+        summarise(`1990_sum` = sum(`1990`)) %>%
+        ungroup ->
+        L100.USSR_Yug_FLOW_PRODUCT
+
+      L100.USSR_Yug_ctry_FLOW_PRODUCT %>%
+        left_join_error_no_match(distinct(select(L100.USSR_Yug_FLOW_PRODUCT, -FLOW), IEAcomposite, PRODUCT, .keep_all = TRUE),
+                                 by = c("IEAcomposite", "PRODUCT")) %>%
+        mutate(`1990_share` = `1990` / `1990_sum`) %>%
+        replace_na(list(`1990_share` = 0)) ->
+        L100.USSR_Yug_ctry_FLOW_PRODUCT_1990
+
+      # Calculate the energy balances of the individual countries during the USSR years as
+      # the total in the composite region times the country-wise shares in 1990 (121-131)
+      POST_USSR_YUG_YEARS_IEA <- intersect(POST_USSR_YUG_YEARS_IEA, names(L100.USSR_Yug_ctry))
+      L100.USSR_Yug_ctry %>%
+        select(one_of("iso", "FLOW", "PRODUCT", "IEAcomposite", POST_USSR_YUG_YEARS_IEA)) %>%
+        left_join_keep_first_only(select(L100.USSR_Yug, one_of("COUNTRY", "FLOW", "PRODUCT", USSR_YUG_YEARS)),
```

Comment:
I feel like it wasn't working because of the integer-ish column names, i.e. years; but will check.